deps/glad/CMakeLists.txt

```diff
@@ -2,29 +2,32 @@ cmake_minimum_required(VERSION 3.22...3.25)
 
 find_package(OpenGL REQUIRED)
 
-add_library(obsglad STATIC EXCLUDE_FROM_ALL )
+add_library(obsglad OBJECT)
 add_library(OBS::glad ALIAS obsglad)
 
 if(OBS_CMAKE_VERSION VERSION_LESS 3.0.0)
   add_library(OBS::obsglad ALIAS obsglad)
 endif()
 
-target_sources(obsglad PRIVATE src/glad.c include/glad/glad.h)
+# cmake-format: off
+target_sources(
+  obsglad
+  PRIVATE src/glad.c
+          $<$<PLATFORM_ID:Windows>:src/glad_wgl.c>
+          $<$<TARGET_EXISTS:OpenGL::EGL>:src/glad_egl.c>
+          $<$<TARGET_EXISTS:OpenGL::EGL>:include/EGL/eglplatform.h>
+  PUBLIC include/glad/glad.h
+         "$<$<PLATFORM_ID:Windows>:${CMAKE_CURRENT_SOURCE_DIR}/include/glad/glad_wgl.h>"
+         "$<$<TARGET_EXISTS:OpenGL::EGL>:${CMAKE_CURRENT_SOURCE_DIR}/include/glad/glad_egl.h>")
+# cmake-format: on
 
-target_include_directories(obsglad PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/include")
-
-target_link_libraries(obsglad PUBLIC OpenGL::GL)
+target_compile_options(obsglad PRIVATE $<$<COMPILE_LANG_AND_ID:C,AppleClang,Clang>:-Wno-strict-prototypes>)
 
-if(OS_WINDOWS)
-  target_sources(obsglad PRIVATE src/glad_wgl.c include/glad/glad_wgl.h)
-elseif(OS_MACOS)
-  target_compile_options(obsglad PRIVATE -Wno-strict-prototypes)
-elseif(OS_LINUX OR OS_FREEBSD)
-  if(TARGET OpenGL::EGL)
-    target_sources(obsglad PRIVATE src/glad_egl.c include/EGL/eglplatform.h include/glad/glad_egl.h)
+target_include_directories(obsglad PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/include")
 
-    target_link_libraries(obsglad PRIVATE OpenGL::EGL)
-  endif()
-endif()
+target_link_libraries(
+  obsglad
+  PRIVATE $<$<NOT:$<PLATFORM_ID:Windows,Darwin>>:${CMAKE_DL_LIBS}>
+  PUBLIC OpenGL::GL $<$<TARGET_EXISTS:OpenGL::EGL>:OpenGL::EGL>)
```

Comment:
This is necessary because `glad` is not a static library anymore, but an `OBJECT` library - it's compiled object files are added directly to the linker command line for `libobs-opengl` and thus its dependencies need to be transitive as well.

Reply:
Got it.